nav2_collision_monitor/include/nav2_collision_monitor/polygon_base.hpp

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2022 Samsung Research Russia
```

Comment:
Just a thought: would it make sense to remove `polygon_base` and have the `base` class be the polygon itself? That is a pretty abstract representation of its own right to be an arbitrary object. Then `circle` is a specialized case of a polygon that could override the appropriate functions. 

I'm looking at polygon / polygon base and seeing enough overlap that I think rather than having a `polygon_base` pure virtual class, we could have the base be itself something of utility and circle be a specialization of it. 

Thoughts?

Reply:
If you agree, also rename `source_base` to just `source` since that'll be the only base class left

Reply:
Basically I am for this idea. Will try it on next update.

Reply:
Done. Merged `PolygonBase` with `Polygon` -> `Polygon`, and renamed `SourceBase` -> to `Source`